include/EntryPointConfirmOptInHandler.php

```diff
@@ -161,11 +163,50 @@ private function methodConfirmOptInUser($request)
         if ($this->emailAddress) {
             $this->emailAddress->confirmOptIn();
             $this->emailAddress->save();
-        }
 
+            $people = $this->getID($this->emailAddress->email_address, 'Contacts');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people, 'Contacts');
+            }
+            $people = $this->getID($this->emailAddress->email_address, 'Leads');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people, 'Leads');
+            }
+
+            $people = $this->getID($this->emailAddress->email_address, 'Prospects');
+            if ($people) {
+                $this->setLawfulBasisForEachPerson($people,  'Prospects');
+            }
+        }
         $template = new Sugar_Smarty();
         $template->assign('FOCUS', $this->emailAddress);
 
         return $template->fetch('include/EntryPointConfirmOptIn.tpl');
     }
+
+    /**
+     * @param String $email
+     * @param String $module
+     *
+     * @return array|bool
+     */
+    private function getID($email, $module) {
```

Comment:
Probably getIDs() in plural would be a more appropriate name since it returns an array.